src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt

```diff
@@ -261,10 +289,56 @@ class ManagedIndexCoordinator(
             if (it.value.shouldDeleteManagedIndexMetaData()) indicesToRemoveManagedIndexMetaDataFrom.add(it.value.index)
         }
 
-        updateManagedIndices(updateManagedIndicesRequests + indicesDeletedRequests, hasCreateRequests)
+        // check if newly created indices matching any ISM templates
+        val updateMatchingIndexReq = getMatchingIndicesUpdateReq(event.state(), event.indicesCreated())
+        if (updateMatchingIndexReq.isNotEmpty()) hasCreateRequests = true
+
+        updateManagedIndices(updateManagedIndicesRequests + updateMatchingIndexReq + indicesDeletedRequests, hasCreateRequests)
```

Comment:
we scan all indices policyID setting before finding matching index, I can record all indices with policyID, remove them from indicesCreated() if exist.

Reply:
Yes, as of now you'll end up with two create requests for the same index right? SInce we're not removing the setting in this PR let's do like you said and make sure the templates skip any indices that have the setting on them (prefer explicitness). I would imagine it's shortly removed after with the follow up PR though which is fine.